### PR TITLE
Update example to use internal FileProvider, autolink works perfectly on RN60

### PR DESCRIPTION
**Pull request by mikehardy**
This is built on #567 and relies on the work from #490 so that autolink works perfectly and should make #549 provably solved, as the example autolinks and works with RN60

If #567 is approved this just has one commit - so this commit is the one to review: https://github.com/react-native-community/react-native-share/pull/570/commits/354bcb464d24a044e761beab572f2ec876536c88

I'll rebase the PR and re-push if/when #567 goes in

# Overview

# Test Plan

### CODE REVIEW

File: example/android/build.gradle [hunk 1]
```diff
@@ -6,14 +6,14 @@ buildscript {
         minSdkVersion = 16
         compileSdkVersion = 28
         targetSdkVersion = 28
-        supportLibVersion = "28.0.0"
+        supportLibVersion = "1.0.0"
```
Comment (MateusAndrade) [inline example/android/build.gradle]: Why did you set this to version 1.0.0? Is this related to Android X?

Comment (mikehardy) [reply example/android/build.gradle]: Yep. RN60 is AndroidX, so your project must go to AndroidX. The required steps in your project are to use Android Studio to [convert your build.gradle dependencies](https://github.com/react-native-community/react-native-share/pull/570/files#diff-15f694d688fe6360b46c5b8399a7eca8R174) and your local native code, and to change your [android.properties](https://github.com/react-native-community/react-native-share/pull/570/files#diff-b5cf0eed029ae59e02a6fb7ff8ef05b9R20) to turn on AndroidX and jetifier. 

Setting the version to 1.0.x here is what corresponds to the version numbers of the AndroidX libraries. 28.x versions are for the old support libraries.

**Comment by mikehardy** [comment]
rebased/pushed after responding to @friederbluemle feedback on #567

**Comment by MateusAndrade** [comment]
Just gave a quick review after the rebase and it looks good to me. IM going to test this today and merge if everything runs fine.

**Comment by MateusAndrade** [comment]
Just tested the example app here and it runs without any problem. If anyone got problems with the example app after this, please let us know. Also, thanks for updating the example app @mikehardy :smile:

**Comment by mikehardy** [comment]
And now my weekly github security alerts email will be (I think) empty :-). Cheers
